packages/expo-updates/src/UseUpdatesUtils.ts

```diff
@@ -0,0 +1,104 @@
+import * as Updates from './Updates';
+import type { UpdateEvent } from './Updates.types';
+import type { AvailableUpdateInfo, UpdatesInfo, UseUpdatesCallbacksType } from './UseUpdates.types';
+import { currentlyRunning } from './UseUpdatesConstants';
+
+/////// Internal functions ////////
+
+// Constructs the availableUpdate from the update manifest
+// Manifest is of type "any" until we no longer support classic updates
+export const availableUpdateFromManifest = (manifest: any) => {
+  return manifest
+    ? {
+        updateId: manifest?.id ?? null,
+        createdAt: manifest?.createdAt
+          ? new Date(manifest?.createdAt)
+          : manifest?.publishedTime
+          ? new Date(manifest?.publishedTime)
+          : null,
+        manifest,
+      }
+    : undefined;
+};
+
+// Constructs the UpdatesInfo from an event
+export const updatesInfoFromEvent = (
+  updatesInfo: UpdatesInfo | undefined,
+  event: UpdateEvent
+): UpdatesInfo => {
+  const lastCheckForUpdateTime = new Date();
+  if (event.type === Updates.UpdateEventType.NO_UPDATE_AVAILABLE) {
```

Comment:
> nit: use a switch to typescript enforces exhaustiveness
>> I will do that in a future PR. Since this PR will be cherry-picked for SDK 48, I'm only supporting the event types that exist there.

Prefer to do the inverse. Do the most correct thing in master (the switch statement), and then during the cherry-pick to the 48 branch you can either remove the other cases from the switch or change it back to an if/else.

Reply:
Fixed